homeassistant/components/neato/vacuum.py

```diff
@@ -34,16 +40,60 @@
 ATTR_CLEAN_BATTERY_END = 'battery_level_at_clean_end'
 ATTR_CLEAN_SUSP_COUNT = 'clean_suspension_count'
 ATTR_CLEAN_SUSP_TIME = 'clean_suspension_time'
+ATTR_MAP_ID = 'map_id'
+
+ATTR_MODE = 'mode'
+ATTR_NAVIGATION = 'navigation'
+ATTR_CATEGORY = 'category'
+ATTR_ZONE = 'zone'
+
+SERVICE_NEATO_CUSTOM_CLEANING = 'neato_custom_cleaning'
+
+SERVICE_NEATO_CUSTOM_CLEANING_SCHEMA = VACUUM_SERVICE_SCHEMA.extend({
+    vol.Optional(ATTR_MODE, default=2): cv.positive_int,
+    vol.Optional(ATTR_NAVIGATION, default=1): cv.positive_int,
+    vol.Optional(ATTR_CATEGORY, default=4): cv.positive_int,
+    vol.Optional(ATTR_ZONE): cv.string
+})
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Neato vacuum."""
+
     dev = []
     for robot in hass.data[NEATO_ROBOTS]:
         dev.append(NeatoConnectedVacuum(hass, robot))
+
     _LOGGER.debug("Adding vacuums %s", dev)
     add_entities(dev, True)
 
+    if dev:
+        def neato_custom_cleaning_service(call):
+            """Zone cleaning service that allows user to change options"""
+            for robot in service_to_entities(call):
+                if call.service == SERVICE_NEATO_CUSTOM_CLEANING:
+                    mode = call.data.get(ATTR_MODE)
+                    navigation = call.data.get(ATTR_NAVIGATION)
+                    category = call.data.get(ATTR_CATEGORY)
+                    zone = call.data.get(ATTR_ZONE)
+                    robot.neato_custom_cleaning(
+                        mode, navigation, category, zone)
+
+        def service_to_entities(call):
+            """Return the known devices that a service call mentions."""
+            entity_ids = extract_entity_ids(hass, call)
+            if entity_ids:
+                entities = [entity for entity in dev
+                            if entity.entity_id in entity_ids]
+            else:
+                entities = None
+
+            return entities
+
+        hass.services.register(DOMAIN, SERVICE_NEATO_CUSTOM_CLEANING,
```

Comment:
Here's the context:
https://github.com/home-assistant/architecture/issues/124#issuecomment-449024744

Reply:
Okay, I'll update that!

Reply:
Done in the latest commit

Reply:
Because the domain is now `neato` and not `vacuum` anymore, the service UI doesn't display automatically the available entities, is this expected? Can I do something to change that behaviour on the component or the service?
![image](https://user-images.githubusercontent.com/21064206/52429191-d6ca7700-2b03-11e9-92b7-5f8f77abe5fe.png)